src/rubrix/client/datasets.py

```diff
@@ -763,7 +763,9 @@ def from_pandas(
         return super().from_pandas(dataframe)
 
     @_requires_datasets
```

Comment:
We should remove this decorator here, since if you choose the `spacy` framework you do not need the *datasets* package. 

I would create two helper methods:

- `_prepare_for_training_with_spacy()`
- `_prepare_for_training_with_transformers()`

and decorate the latter one with `_requires_datasets`, and create a similar `_requires_spacy` for the former.
So the `prepare_for_training` method just does the value comparison of the `framework` argument and calls the corresponding helper method.